docs/reference/how-to/search-speed.asciidoc

```diff
@@ -164,8 +164,163 @@ include::../mapping/types/numeric.asciidoc[tag=map-ids-as-keyword]
 [float]
 === Avoid scripts
 
-In general, scripts should be avoided. If they are absolutely needed, you
-should prefer the `painless` and `expressions` engines.
+If possible, avoid using <<modules-scripting,scripts>> or
+<<request-body-search-script-fields,scripted fields>> in searches. Using scripts
+in search queries can result in slower search speeds.
```

Comment:
I'd suggest adding something like "because they can't make use of index structures" to emphasize that scripts are not intrisically slow, it's just that they prevent optimizations.

Reply:
Thanks @jpountz. Added with 67182f180c5.